test/pool-shift-async/test.c

```diff
@@ -97,7 +108,10 @@ child_main(void) {
   mtev_dso_post_init();
 
   mtev_conf_write_log();
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcast-function-type"
```

Comment:
don't want pragmas like this in the code, please provide the error message, should be able to either add the appropriate cast (or remove it)

Reply:
This is what happens without the pragma:
```
pool-shift-async/test.c: In function ‘child_main’:
pool-shift-async/test.c:111:40: error: cast between incompatible function types from ‘int (*)(void)’ to ‘int (*)(void *)’ [-Werror=cast-function-type]
  111 |   mtev_conf_watch_and_journal_watchdog((int (*)(void *))mtev_conf_write_log, NULL);
      |                                        ^
cc1: all warnings being treated as errors
make: *** [Makefile:60: pool-shift-async/test] Error 1
```

Reply:
Offline we decided to create a new function with the correct signature and forwarding from there.